### ISSUE TITLE
Interacting with QT : Graphics are not displaying until clicking on graph area

### ISSUE THREAD

**Issue by MYOSUN** [pyvista/pyvista-support#86]
Dears,

I'm using QT designer to create UI. After conversion from .ui to .py, i'm editing the code to interact with pyvista. But the graphics are not displaying directly.  I shared the simple code. When i click to ADD BOX button the graph is not displaying until i click on graph area. Could you please help me to handle this problem.

```py
from PyQt5 import QtCore, QtGui, QtWidgets
import pyvista as pv

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(792, 588)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayoutWidget = QtWidgets.QWidget(self.centralwidget)
        self.verticalLayoutWidget.setGeometry(QtCore.QRect(220, 10, 561, 531))
        self.verticalLayoutWidget.setObjectName("verticalLayoutWidget")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.verticalLayoutWidget)
        self.verticalLayout.setContentsMargins(0, 0, 0, 0)
        self.verticalLayout.setObjectName("verticalLayout")
		
        
        # add the pyvista interactor object
        self.vtk_widget = pv.QtInteractor(self.verticalLayoutWidget)
        self.verticalLayout.addWidget(self.vtk_widget)
		

        self.add_box_Button = QtWidgets.QPushButton(self.centralwidget)
        self.add_box_Button.setGeometry(QtCore.QRect(10, 10, 201, 61))
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.add_box_Button.setFont(font)
        self.add_box_Button.setObjectName("add_box_Button")
        self.clear_Button = QtWidgets.QPushButton(self.centralwidget)
        self.clear_Button.setGeometry(QtCore.QRect(10, 80, 201, 61))
		
        #add action for BUTTONs
        self.add_box_Button.clicked.connect(self.add_box)
        self.clear_Button.clicked.connect(self.clear)
		
		
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.clear_Button.setFont(font)
        self.clear_Button.setObjectName("clear_Button")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(10, 150, 201, 31))
        font = QtGui.QFont()
        font.setPointSize(16)
        self.label.setFont(font)
        self.label.setText("")
        self.label.setObjectName("label")
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 792, 21))
        self.menubar.setObjectName("menubar")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.add_box_Button.setText(_translate("MainWindow", "ADD BOX"))
        self.clear_Button.setText(_translate("MainWindow", "CLEAR"))

    def add_box(self):
        box=pv.Box()
        self.vtk_widget.add_mesh(box)
        self.vtk_widget.reset_camera()
        self.label.setText('BOX ADDED')
       
    def clear(self):
        self.vtk_widget.clear()
        self.label.setText('CLEARED')

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
```

**Comment by banesullivan** [comment]
So a quick fix is to use the plotter's `update()` method after updating the scene.

```py
...
    def add_box(self):
        box=pv.Box()
        self.vtk_widget.add_mesh(box)
        self.vtk_widget.reset_camera()
        self.vtk_widget.update() #  <----------- this updates the widget
        self.label.setText('BOX ADDED')
```

Honestly, I'm not sure why this is needed here... it should be updating. @akaszynski is the most knowledgable with Qt and PyVista

**Comment by banesullivan** [comment]
Also, thanks for sharing this snippet - I may add something similar in the docs to show how to use PyVista in a Qt app

**Comment by akaszynski** [comment]
I've noticed this behavior as well.  Adding, removing, and modifying objects within a Qt app's VTK widget need to trigger a manual update despite the fact that this should occur automatically.  In some of my closed source work I've implemented a wrapper around the `add_mesh` method to automatically update the widget, and this might be something we should do with the `QtInteractor`.

It should be noted that pressing "q" in this Qt app destroys the VTK widget without shutting down the application.  We might want to override this behavior by default...
